### ISSUE TITLE
Internal Server error on form mutation

### ISSUE THREAD

**Issue by daltonrooney** [verbb/formie#416]
**Description**

A previously working mutation is now causing an internal error. 

```
Argument 1 passed to craft\\services\\Elements::getElementById() must be of the type int, null given, called in /home/site/webapps/api-staging/releases/faefc07d9958b19b07c31bece3b96223ffbdae0a/craft/vendor/verbb/formie/src/gql/resolvers/mutations/SubmissionResolver.php on line 138"
```

### Form settings

- Submission Method: GraphQL mutation

**Additional info**
- Plugin version: 1.3.25
- Craft version: 3.6.11.2
- Multi-site: No

**Additional context**

Example of the mutation

```
{"query":"    mutation formSubmission($name1: contact_name1_FormieNameInput,$role: String,$address: contact_address_FormieAddressInput,$email: String,$message: String) {      save_contact_Submission(name1: $name1,role: $role,address: $address,email: $email,message: $message) {        title        name1 role address email message       }    }  ","variables":{"name1":{"firstName":"Dalton","lastName":"Rooney"},"role":"communications@redacted.org","email":"dalton@redacted.com","message":"asdfasdf","address":{"address1":"","city":"","state":"","zip":""}}}
```

Full response: 
```
{
	"errors": [
		{
			"debugMessage": "Argument 1 passed to craft\\services\\Elements::getElementById() must be of the type int, null given, called in /home/site/webapps/api-staging/releases/faefc07d9958b19b07c31bece3b96223ffbdae0a/craft/vendor/verbb/formie/src/gql/resolvers/mutations/SubmissionResolver.php on line 138",
			"message": "Internal server error",
			"extensions": {
				"category": "internal"
			},
			"file": "/home/site/webapps/api-staging/releases/faefc07d9958b19b07c31bece3b96223ffbdae0a/craft/vendor/craftcms/cms/src/services/Elements.php",
			"line": 456,
			"trace": [
				{
					"file": "/home/site/webapps/api-staging/releases/faefc07d9958b19b07c31bece3b96223ffbdae0a/craft/vendor/verbb/formie/src/gql/resolvers/mutations/SubmissionResolver.php",
					"line": 138,
					"call": "craft\\services\\Elements::getElementById()"
				},
				{
					"file": "/home/site/webapps/api-staging/releases/faefc07d9958b19b07c31bece3b96223ffbdae0a/craft/vendor/webonyx/graphql-php/src/Executor/ReferenceExecutor.php",
					"line": 624,
					"call": "verbb\\formie\\gql\\resolvers\\mutations\\SubmissionResolver::saveSubmission()"
				},
				{
					"file": "/home/site/webapps/api-staging/releases/faefc07d9958b19b07c31bece3b96223ffbdae0a/craft/vendor/webonyx/graphql-php/src/Executor/ReferenceExecutor.php",
					"line": 550,
					"call": "GraphQL\\Executor\\ReferenceExecutor::resolveFieldValueOrError()"
				},
				{
					"file": "/home/site/webapps/api-staging/releases/faefc07d9958b19b07c31bece3b96223ffbdae0a/craft/vendor/webonyx/graphql-php/src/Executor/ReferenceExecutor.php",
					"line": 474,
					"call": "GraphQL\\Executor\\ReferenceExecutor::resolveField()"
				},
				{
					"file": "/home/site/webapps/api-staging/releases/faefc07d9958b19b07c31bece3b96223ffbdae0a/craft/vendor/webonyx/graphql-php/src/Executor/ReferenceExecutor.php",
					"line": 858,
					"call": "GraphQL\\Executor\\ReferenceExecutor::GraphQL\\Executor\\{closure}()"
				},
				{
					"call": "GraphQL\\Executor\\ReferenceExecutor::GraphQL\\Executor\\{closure}()"
				},
				{
					"file": "/home/site/webapps/api-staging/releases/faefc07d9958b19b07c31bece3b96223ffbdae0a/craft/vendor/webonyx/graphql-php/src/Executor/ReferenceExecutor.php",
					"line": 860,
					"function": "array_reduce()"
				},
				{
					"file": "/home/site/webapps/api-staging/releases/faefc07d9958b19b07c31bece3b96223ffbdae0a/craft/vendor/webonyx/graphql-php/src/Executor/ReferenceExecutor.php",
					"line": 490,
					"call": "GraphQL\\Executor\\ReferenceExecutor::promiseReduce()"
				},
				{
					"file": "/home/site/webapps/api-staging/releases/faefc07d9958b19b07c31bece3b96223ffbdae0a/craft/vendor/webonyx/graphql-php/src/Executor/ReferenceExecutor.php",
					"line": 263,
					"call": "GraphQL\\Executor\\ReferenceExecutor::executeFieldsSerially()"
				},
				{
					"file": "/home/site/webapps/api-staging/releases/faefc07d9958b19b07c31bece3b96223ffbdae0a/craft/vendor/webonyx/graphql-php/src/Executor/ReferenceExecutor.php",
					"line": 215,
					"call": "GraphQL\\Executor\\ReferenceExecutor::executeOperation()"
				},
				{
					"file": "/home/site/webapps/api-staging/releases/faefc07d9958b19b07c31bece3b96223ffbdae0a/craft/vendor/webonyx/graphql-php/src/Executor/Executor.php",
					"line": 156,
					"call": "GraphQL\\Executor\\ReferenceExecutor::doExecute()"
				},
				{
					"file": "/home/site/webapps/api-staging/releases/faefc07d9958b19b07c31bece3b96223ffbdae0a/craft/vendor/webonyx/graphql-php/src/GraphQL.php",
					"line": 162,
					"call": "GraphQL\\Executor\\Executor::promiseToExecute()"
				},
				{
					"file": "/home/site/webapps/api-staging/releases/faefc07d9958b19b07c31bece3b96223ffbdae0a/craft/vendor/webonyx/graphql-php/src/GraphQL.php",
					"line": 94,
					"call": "GraphQL\\GraphQL::promiseToExecute()"
				},
				{
					"file": "/home/site/webapps/api-staging/releases/faefc07d9958b19b07c31bece3b96223ffbdae0a/craft/vendor/craftcms/cms/src/services/Gql.php",
					"line": 545,
					"call": "GraphQL\\GraphQL::executeQuery()"
				},
				{
					"file": "/home/site/webapps/api-staging/releases/faefc07d9958b19b07c31bece3b96223ffbdae0a/craft/vendor/craftcms/cms/src/controllers/GraphqlController.php",
					"line": 172,
					"call": "craft\\services\\Gql::executeQuery()"
				},
				{
					"call": "craft\\controllers\\GraphqlController::actionApi()"
				},
				{
					"file": "/home/site/webapps/api-staging/releases/faefc07d9958b19b07c31bece3b96223ffbdae0a/craft/vendor/yiisoft/yii2/base/InlineAction.php",
					"line": 57,
					"function": "call_user_func_array()"
				},
				{
					"file": "/home/site/webapps/api-staging/releases/faefc07d9958b19b07c31bece3b96223ffbdae0a/craft/vendor/yiisoft/yii2/base/Controller.php",
					"line": 181,
					"call": "yii\\base\\InlineAction::runWithParams()"
				},
				{
					"file": "/home/site/webapps/api-staging/releases/faefc07d9958b19b07c31bece3b96223ffbdae0a/craft/vendor/craftcms/cms/src/web/Controller.php",
					"line": 190,
					"call": "yii\\base\\Controller::runAction()"
				},
				{
					"file": "/home/site/webapps/api-staging/releases/faefc07d9958b19b07c31bece3b96223ffbdae0a/craft/vendor/yiisoft/yii2/base/Module.php",
					"line": 534,
					"call": "craft\\web\\Controller::runAction()"
				},
				{
					"file": "/home/site/webapps/api-staging/releases/faefc07d9958b19b07c31bece3b96223ffbdae0a/craft/vendor/craftcms/cms/src/web/Application.php",
					"line": 274,
					"call": "yii\\base\\Module::runAction()"
				},
				{
					"file": "/home/site/webapps/api-staging/releases/faefc07d9958b19b07c31bece3b96223ffbdae0a/craft/vendor/yiisoft/yii2/web/Application.php",
					"line": 104,
					"call": "craft\\web\\Application::runAction()"
				},
				{
					"file": "/home/site/webapps/api-staging/releases/faefc07d9958b19b07c31bece3b96223ffbdae0a/craft/vendor/craftcms/cms/src/web/Application.php",
					"line": 259,
					"call": "yii\\web\\Application::handleRequest()"
				},
				{
					"file": "/home/site/webapps/api-staging/releases/faefc07d9958b19b07c31bece3b96223ffbdae0a/craft/vendor/yiisoft/yii2/base/Application.php",
					"line": 392,
					"call": "craft\\web\\Application::handleRequest()"
				},
				{
					"file": "/home/site/webapps/api-staging/releases/faefc07d9958b19b07c31bece3b96223ffbdae0a/public/index.php",
					"line": 21,
					"call": "yii\\base\\Application::run()"
				}
			]
		}
	],
	"data": {
		"save_contact_Submission": null
	}
}
```

**Comment by engram-design** [comment]
Can't seem to replicate this one, with your field setup (Name, Address, Email, etc), and I believe only Name and Address have been updated recently, as they weren't working correctly with GQL.

However, this error seems to point to an un-saved submission which could mean any number of things. Is there nothing in the logs files?

I've just pushed a minor update to `craft-3` to catch this error and hopefully provide some insight as to why the submission doesn't save.
